rpm_s3_mirror/repository.py

```diff
@@ -157,6 +160,13 @@ def parse_metadata(self) -> Metadata:
         package_list = self._extract_package_list(primary=repodata["primary"])
         return Metadata(package_list=package_list, repodata=repodata, base_url=self.base_url)
 
+    def exists(self):
+        try:
+            self._req(self.session.get, "repodata/repomd.xml")
+        except HTTPError:
```

Comment:
Should probably check for 404 for not present, and 200 for present and raise on anything else.

Reply:
Fixed, thank you. S3 will respond with HTTP 403 (Access Denied) if the object does not exist. Also check HTTP 404 for normal HTTP server behaviour.